### ISSUE TITLE
node_exporter cannot use the systemd collector on api-servers due to AppArmor policies

### ISSUE THREAD

**Issue by nkinkade** [m-lab/k8s-support#502]
All api-servers logs are filled with message like following every 10 or 15 seconds:

```
audit: type=1107 audit(1603147863.203:9141): pid=476 uid=105 auid=4294967295 ses=4294967295 msg='apparmor="DENIED" operation="dbus_method_call"  bus="system" path="/org/freedesktop/DBus" interface="org.freedesktop.DBus" member="Hello" mask="send" name="org.freedesktop.DBus" pid=1416615 label="docker-default" peer_label="unconfined"
                 exe="/usr/bin/dbus-daemon" sauid=105 hostname=? addr=? terminal=?'
```

It turns out that this is node_exporter trying to use the `systemd` collector, and failing. Inside of the node_exporter container are corresponding logs that look like:

```
level=error ts=2020-10-19T22:53:03.208Z caller=collector.go:161 msg="collector failed" name=systemd duration_seconds=0.001512062 err="couldn't get dbus connection: An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\"(null)\" (inactive) interface=\"org.freedesktop.DBus\" member=\"Hello\" error name=\"(unset)\" requested_reply=\"0\" destination=\"org.freedesktop.DBus\" (bus)"
```

Presumably we will need to modify whichever AppArmor profile is causing this.

**Comment by nkinkade** [comment]
https://docs.docker.com/engine/security/apparmor/
https://github.com/moby/moby/blob/master/profiles/apparmor/template.go

**Comment by nkinkade** [comment]
@robertodauria: A little over a year ago [you enabled the `systemd` collector in node_exporter](https://github.com/m-lab/k8s-support/pull/317). Are those metrics being used anywhere, or were you just adding it with the intention that it _might_ be used? For some reason the `docker-default` (or for containerd `containerd.apparmor.d`) AppArmor profiles are blocking node_exporter from accessing the dbus socket. I don't know why yet, but it seems that there are no active AppArmor profiles on platform nodes, but there are on API nodes, and node_exporter is not working for the `systemd` collector there.

**Comment by robertodauria** [comment]
@nkinkade IIRC, the intention was to be able to list failed systemd services by looking at metrics such as `node_systemd_system_running` (for a summary) or `node_systemd_units` for a detail of which services have failed. This would allow us to monitor things I don't think we are monitoring otherwise, e.g. if the oneshot service that configures bandwidth limits for a single connection according to the uplink has been run successfully when booting the node.

The idea had some potential but these metrics are not shown on any dashboard nor used in any alert. I made sure the two we wanted are scraped from the platform cluster [here](https://github.com/m-lab/prometheus-support/blob/85b36f83c111a6f04237c033c7689d50c0f0953b/config/federation/prometheus/prometheus.yml.template#L352) but it seems I've never used them.

**Comment by stephen-soltesz** [comment]
Which version does this error occur with? I tried updating sandbox to use v1.1.0 and searched for this error on google only to find this issue. Did this occur with the previous version of node-exporter? (I don't see it in logs of pods still running the old version).

**Comment by nkinkade** [comment]
This issue will be resolved by #551, in which the node_exporter DaemonSet is limited to physical nodes where this problem does not occur. Closing.
